### DISCUSSION TITLE
How can I enhance fuse mount performance?

### DISCUSSION THREAD

**Discussion by sgonv** [seaweedfs/seaweedfs#5089] [Q&A]
I have 1 master, 1 filer with redis2, 28 volume servers with HDD. All server processes are running in the same machine. (I have one server hardware with 28 HDD)
I use seaweedfs with fuse mount in my personal mac, but it seems that its performance is not good.
I compared my benchmark with https://github.com/seaweedfs/seaweedfs/wiki/FUSE-Mount#sysbench-benchmark-results
my benchmark has lower performance than above link's performance.
```
File operations:
    reads/s:                      32.13
    writes/s:                     21.42
    fsyncs/s:                     69.07

Throughput:
    read, MiB/s:                  32.13
    written, MiB/s:               21.42

General statistics:
    total time:                          61.6179s
    total number of events:              7428

Latency (ms):
         min:                                    0.02
         avg:                                    8.08
         max:                                  448.21
         95th percentile:                        0.00
         sum:                                60026.40
```
I think file operations and throughput should be higher than this performance. (Also I think max latency is little bit suspicious...)
Do you think this performance is reasonable? and is there any way to enhance the performance of fuse mount?

**Comment by n9** [comment]
How the two computers are connected? What about some network benchmarks?

**Reply by sgonv** [reply]
Even though I mounted the filer in the same machine, it's performance is similar to my local machine. Not network issue.

**Reply by sgonv** [reply]
```
$ sysbench --test=fileio --file-total-size=1G prepare
$ sysbench --test=fileio --file-total-size=1G --file-test-mode=rndrw --max-time=60 --max-requests=0 --num-threads=1 --file-block-size=1m run
```
I used these commands for my benchmark test. So I expected that benchmark is similar to wiki's benchmark, but it was not.
Also, actual file read is very slow. it takes 50ms to read only 4bytes. I think it is totally not reasonable, and there might be some problem in my configuration.

**Reply by n9** [reply]
What OS you have on the server?

**Reply by sgonv** [reply]
Ubuntu 22.04.3

**Reply by M1Sports20** [reply]
@sgonv Were you able to figure out the source for the slow bandwidth?  I am having almost the exact same problem and speeds as you when I know I should not be.
